packages/utils/src/schema/getDefaultFormState.ts

```diff
@@ -314,7 +325,49 @@ export function computeDefaults<T = any, S extends StrictRJSFSchema = RJSFSchema
 
   const defaultBasedOnSchemaType = getDefaultBasedOnSchemaType(validator, schema, computeDefaultsProps, defaults);
 
-  return defaultBasedOnSchemaType ?? defaults;
+  let defaultsWithFormData = defaultBasedOnSchemaType ?? defaults;
+  // if shouldMfMergeDefaultsIntoFormData is true, then merge the defaults into the formData.
+  if (shouldMergeDefaultsIntoFormData) {
+    const { arrayMinItems = {} } = experimental_defaultFormStateBehavior || {};
+    const { mergeExtraDefaults } = arrayMinItems;
+
+    const validFormData = getValidFormData(validator, schema, rootSchema, rawFormData);
+    if (!isObject(rawFormData)) {
+      defaultsWithFormData = mergeDefaultsWithFormData<T>(
+        defaultsWithFormData as T,
+        validFormData as T,
+        mergeExtraDefaults,
+        true
+      ) as T;
+    }
+  }
+
+  return defaultsWithFormData;
+}
+
+/**
+ * Gets valid formData. If it's not valid in the case of a selectField, we change it to a valid value.
+ * @param validator - an implementation of the `ValidatorType` interface that will be used when necessary
+ * @param schema - The schema for which the formData state is desired
+ * @param rootSchema The root schema, used to primarily to look up `$ref`s
+ * @param formData The current formData
+ * @returns valid formData
+ */
+export function getValidFormData<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>(
```

Comment:
This doesn't really check validity w.r.t. to the schema so I think this name might be confusing. Maybe `getFormDataMatchingSchema`?

Reply:
Hi @nickgros, I think it's a much better name but I've changed it to **`ensureFormDataMatchingSchema`** because I think it's slightly better.